src/System.Text.Json/tests/WritableJsonApiTests.cs

```diff
@@ -625,5 +660,187 @@ public static void TestAquiringAllProperties()
+                    ""age"" : 24,                
+                }
+            }";
+
+            JsonObject employees = JsonNode.Parse(jsonString) as JsonObject;
+            employees.Add(EmployeesDatabase.GetNextEmployee());
+            Mailbox.SendAllEmployeesData(JsonDocument.Parse(employees));
+        }
+
+        /// <summary>
+        /// Copying JsoneNode
+        /// </summary>
+        [Fact]
+        public static void TestCopyingJsonNode()
+        {
+            JsonObject employee = EmployeesDatabase.GetManager();
+            JsonNode employeeCopy = JsonElement.DeepCopy(employee);
+        }
+
+        /// <summary>
+        /// Checking IsImmutable property
+        /// </summary>
+        [Fact]
+        public static void TestIsImmutable()
+        {
+            JsonElement employee = Mailbox.RetrieveEmployeeData();
+            if(!employee.IsImmutable)
+            {
+                employee = JsonDocument.Parse(employee).RootElement;
+            }
+
+            // Perform operations requiring high performance
+            EmployeesDatabase.PerformHeavyOperations(employee);
+        }
+
+        /// <summary>
+        /// Sorting string JsonArray
+        /// </summary>
+        [Fact]
+        public static void TestSortingJsonArray()
+        {
+            var employees = new JsonArray
+            {
+                "Smith John",
+                "Predictable Ann",
+                "Coder Zoe",
+                "Shadow Byron"
+            };
+
+            employees.Sort();
+            
+            // Create medical appointments for employees in alphabetical order
+            foreach (JsonNode employee in employees)
+            {
+                if(employee is JsonString person)
+                    HealthCare.CreateMedicalAppointment(person.Value);
+            }
+        }
+
+        /// <summary>
+        /// Sorting multi-typed JsonArray
+        /// </summary>
+        [Fact]
+        public static void TestSortingMultiTypedJsonArray()
+        {
+            var studentGrades = new JsonArray
+            {
+                3,
+                5,
+                4,
+                "not passed",
+                5,
+                3.5,
+                "passed with no grade",
+                "passed with no grade"
+            };
+
+            studentGrades.Sort();
+
+            // Calculate median of grades:
+
+            int lastIdx = 0;
+            while (studentGrades[lastIdx] is JsonNumber)
+                lastIdx++;
+
+            double median;
+
+            if(lastIdx % 2 == 0)
+            {
+                median = ((JsonNumber)studentGrades[lastIdx/2]).GetInt32();
+            }
+            else
+            {
+                double left = ((JsonNumber)studentGrades[lastIdx/2]).GetDouble();
+                double right = ((JsonNumber)studentGrades[lastIdx/2 + 1]).GetDouble();
+                median = (left + right) / 2.0;
+            }
```

Comment:
It's unclear to me what the value is of any of the code after the call to Sort is...

Reply:
Just to show why someone would want to sort it (as I wrote, those are supposed to be scenarios not test, I assumed I should make them real-life examples).

Reply:
I'd probably write this as something like

```C#
[Fact]
public static void SortMultiTypedArray()
{
    var studentGrades = new JsonArray
    {
        3,
        5,
        4,
        "not passed",
        5,
        3.5,
        "passed with no grade",
        "passed with no grade"
    };

    double ComputeMedian(JsonArray grades)
    {
        if ((grades?.Length ?? 0) == 0)
        {
            return double.NaN;
        }

        grades.Sort();

        // [median computation goes here, maybe checks that there were no numeric grades at all]
        return median;
    }

    Assert.Equals(4, ComputeMedian(studentGrades));
    // ComputeMedian sorted the array, the side effect has leaked out:
    Assert.Equals(3.5, ((JsonNumber)studentGrades[1]).GetDouble());
}
```

That way it shows the setup, the scenario code (the local method) and the verification.

Reply:
Thanks! I'll change it.